### PR TITLE
Issue #531: GitHub create-pr not working on remotes with ssh prefix

### LINKED ISSUE #531: GitHub create-pr not working on remotes with ssh prefix
```sh
mkondratek@VL-D-1096 git-machete-intellij-plugin % git machete github create-pr
Remotes are defined for this repository, but none of them corresponds to GitHub (see git remote -v for details)

mkondratek@VL-D-1096 git-machete-intellij-plugin % git remote -v
origin  ssh://git@github.com/VirtusLab/git-machete-intellij-plugin.git (fetch)
origin  ssh://git@github.com/VirtusLab/git-machete-intellij-plugin.git (push)
```

### PR DESCRIPTION
**Pull request by amalota**

### CODE REVIEW

File: git_machete/github.py [hunk 1]
```diff
@@ -23,6 +23,7 @@
 GITHUB_REMOTE_PATTERNS = [
     "^https://github\\.com/(.*)/(.*)\\.git$",
     "^git@github\\.com:(.*)/(.*)\\.git$",
+    "^ssh://git@github\\.com/(.*)/(.*)\\.git$"
```
Comment (PawelLipski) [inline git_machete/github.py:26]: Or just '(ssh://)?'

Comment (amalota) [reply git_machete/github.py:26]: @PawelLipski Why?

Comment (PawelLipski) [reply git_machete/github.py:26]: Sorry, I meant:
```python
GITHUB_REMOTE_PATTERNS = [
    "^https://github\\.com/(.*)/(.*)\\.git$",
    "^(ssh://)?git@github\\.com/(.*)/(.*)\\.git$"
]
```

Just to make it clear that it's about a "flavor" of the "classic" SSH URL, not some another distinct URL type

Comment (amalota) [reply git_machete/github.py:26]: @PawelLipski `ssh://` isn't the only difference, `git@github\\.com:` changes into `git@github\\.com/` (`:` after `.com` into `/`), so the regex would have to be `^(?:ssh:\/\/)?git@github\.com(?:\/|:)(.*)\/(.*)\.git$` (`(?:X)` is needed to create non-capturing group so we can always expect that only 2 groups will be present in the match as its expected in:

```
def get_parsed_github_remote_url(url: str, remote: str) -> Optional[RemoteAndOrganizationAndRepository]:

    for pattern in GITHUB_REMOTE_PATTERNS:
        match = re.match(pattern, url)
        if match:
            return RemoteAndOrganizationAndRepository(remote=remote,
                                                      organization=match.group(1),
                                                      repository=match.group(2))
    return None

File: git_machete/github.py [hunk 2]
```diff
@@ -22,7 +22,7 @@
 GITHUB_DOMAIN = "github.com"
 GITHUB_REMOTE_PATTERNS = [
     "^https://github\\.com/(.*)/(.*)\\.git$",
-    "^git@github\\.com:(.*)/(.*)\\.git$",
+    "^(?:ssh:\\/\\/)?git@github\\.com(?:\\/|:)(.*)/(.*)\\.git$"
```
Comment (PawelLipski) [inline git_machete/github.py]: Ouch, sorry, I didn't notice the subtle difference other than `ssh://` prefix... also forgot about the non-capturing groups

Comment (PawelLipski) [reply git_machete/github.py]: So indeed, if you could revert your original version, that was much more readable indeed!

File: git_machete/github.py (hunk 1 above)
Comment (PawelLipski) [reply git_machete/github.py:26]: (resolved in favor of keeping 2 separate regexes)

File: RELEASE_NOTES.md [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 ## New in git-machete 3.10.1
 
+- added: support GitHub remote url in the form of `ssh://git@github.com/USERNAME/REPOSITORY.git`
```
Comment (PawelLipski) [inline RELEASE_NOTES.md]:
```suggestion
- added: support GitHub remote URL in the form of `ssh://git@github.com/USERNAME/REPOSITORY.git`
```

**Review by PawelLipski — APPROVED** [review]

**Pull request merged**